include/deal.II/base/patterns.h

```diff
@@ -1265,7 +1265,43 @@ namespace Patterns
                           Convert<T>::to_pattern()) = delete;
     };
 
+    /**
+     * A utility function that simplifies the conversion to strings of arbitrarily
+     * complex types.
+     *
+     * This function calls the method Convert<T>::to_string() with the default
+     * pattern. An example usage is the following:
+     *
+     * @code
+     * auto t = std::make_tuple(1.0, std::make_pair(1, "ciao"));
+     * auto s = Patterns::Tools::to_string(t);
+     *
+     * std::cout << s; // will print "1 % 1 : ciao
+     * @endcode
+     *
+     * @author Luca Heltai, 2018
+     */
+    template<typename T>
+    std::string to_string(const T &t);
 
+    /**
+     * A utility function that simplifies the conversion from strings to arbitrary
+     * types.
+     *
+     * This function calls the method Convert<T>::to_value() with the default
+     * pattern. An example usage is the following:
+     *
+     * @code
+     * auto t = std::make_tuple(1.0, std::make_pair(1, "ciao"));
+     * to_value("2 % 3 : mondo", t);
+     * auto s = Patterns::Tools::to_string(t);
+     * std::cout << s; // will print "2 % 3 : mondo
+     * @endcode
```

Comment:
Its not directly related to this PR, but where do we document usage of `%` and `:`? I cannot find it.